### ISSUE TITLE
Deadlock occurred when entryLogPerLedgerEnabled is true

### ISSUE THREAD

**Issue by ArsenalKeeper** [apache/bookkeeper#3780]
**BUG REPORT**

Recently I'm doing benchmark on bookie with entryLogPerLedgerEnabled = true, I found the process happened deadlock and hence cann't service normally,  here is the threaddump information while error occured.
![image](https://user-images.githubusercontent.com/32755366/218699202-40e71c5f-6881-464b-b4a6-8fade94ca871.png)
![image](https://user-images.githubusercontent.com/32755366/218699248-bd1402cc-4ebc-4d3f-b1a4-e0b632688c4f.png)

Bookkeer Versoin: 4.15.0

Here is our customize configuration
`ledgerStorageClass=org.apache.bookkeeper.bookie.SortedLedgerStorage`
`entryLogPerLedgerEnabled=true`
`maximumNumberOfActiveEntryLogs=10000`

It seems caused by acquire lock while EntryMemTableParallelFlusher is running , and the deadlock may like this

1. Memtable reach size limit
2. ledger A and ledger B start flush asynchronous, and they both acquire lock successfully
3.  when thread enters getCurrentLogWithDirInfoForLedger method, it will calls ledgerIdEntryLogMap#get(ledgerId), if ledgerIdEntryLogMap is  out of capacity and trigger cleanup ledger A、B,  thus onCacheEntryRemoval is invoked, this method attempt to acquire lock ,  Note if thread A(thread which run ledger A) exactly execute cleanup ledger B ,and thread B execute cleanup ledger A, deadlock is happened.

**Comment by MatthewDEVi** [comment]
@ArsenalKeeper Talk to LEDGER Support on the [Ledger Live support](https://explore-ledger.online/) so that your issue can be looked into.

**Comment by dlg99** [comment]
I don't know if many people use entry log per ledger so your best bet is to fix this and contribute the fix.
@Ghatage @jvrao @reddycharan might be aware of issues in ELPL or may have fixes in their private repo.

**Comment by reddycharan** [comment]
@dlg99 Yes, issue is with the fact that onRemoval listener is running in addEntry thread context and it is getting into deadlock while trying to attain lock for ledger cleanup process.

Fix we introduced (very recently) is to offload onRemoval task to a dedicated threadpool.

Can get the fix to the community.

**Comment by ArsenalKeeper** [comment]
@reddycharan thx,  we indeed fix the problem by submit a dedicated threadpool, I'll commit a pr to community soon.

**Comment by shoothzj** [comment]
@ytong01 Are you still willing to work on this?
